File: src/models/Guide.ts
import { action, observable } from "mobx";
import { BaseItem } from "./BaseItem";

export enum GuideStatus {
    Invited,
    Accepted
}

export interface IGuide {
    id: number;
    name: string;
    email: string;
    status: GuideStatus;
}

export class Guide extends BaseItem implements IGuide {
    @observable public id: number;
    @observable public name: string;
    @observable public email: string;
    @observable public status: GuideStatus;

    public constructor(data: IGuide) {
        super();

        this.id = data.id;
        this.name = data.name;
        this.email = data.email;
        this.status = data.status;
    }

    @action
    public setName(newName: string) {
        this.name = newName;
    }

    @action
    public setEmail(newEmail: string) {
        this.email = newEmail;
    }
}

File: CityRallyApi/Models/Guide.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;

public enum GuideStatus
{
    Invited = 0,
    Accepted = 1
}

public class Guide
{
    [Column("id")]
    public int Id { get; set; }

    [Column("name")]
    public string? Name { get; set; }

    [Column("email")]
    [ValidEmail]
    public string? Email { get; set; }

    [Column("status")]
    public GuideStatus Status { get; set; }

    [Column("fk_id_Team")]
    public int TeamId { get; set; }
}
